bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/classic/BigtableHBaseClassicSettings.java

```diff
@@ -0,0 +1,348 @@
+      bigtableOptionsBuilder.setDataHost(dataHostOverride);
+    }
+
+    String adminHostOverride = configuration.get(BigtableOptionsFactory.BIGTABLE_ADMIN_HOST_KEY);
+    if (adminHostOverride != null) {
+      LOG.debug("Admin endpoint host %s.", adminHostOverride);
+      bigtableOptionsBuilder.setAdminHost(adminHostOverride);
+    }
+
+    String portOverrideStr = configuration.get(BigtableOptionsFactory.BIGTABLE_PORT_KEY);
+    if (portOverrideStr != null) {
+      bigtableOptionsBuilder.setPort(Integer.parseInt(portOverrideStr));
+    }
+
+    String usePlaintextStr = configuration.get(BIGTABLE_USE_PLAINTEXT_NEGOTIATION);
+    if (usePlaintextStr != null) {
+      bigtableOptionsBuilder.setUsePlaintextNegotiation(Boolean.parseBoolean(usePlaintextStr));
+    }
+
+    setBulkOptions(bigtableOptionsBuilder);
+    setChannelOptions(bigtableOptionsBuilder);
+    setClientCallOptions(bigtableOptionsBuilder);
+
+    String emulatorHost = configuration.get(BigtableOptionsFactory.BIGTABLE_EMULATOR_HOST_KEY);
+    if (emulatorHost != null) {
+      bigtableOptionsBuilder.enableEmulator(emulatorHost);
+    }
+
+    String useBatchStr = configuration.get(BIGTABLE_USE_BATCH);
+    if (useBatchStr != null) {
+      bigtableOptionsBuilder.setUseBatch(Boolean.parseBoolean(useBatchStr));
+    }
+
+    String useGcjClientStr = configuration.get(BIGTABLE_USE_GCJ_CLIENT);
+    if (useGcjClientStr != null) {
+      bigtableOptionsBuilder.setUseGCJClient(Boolean.parseBoolean(useGcjClientStr));
+    }
+    return bigtableOptionsBuilder.build();
+  }
+  // </editor-fold>
+
+  // <editor-fold desc="Private Helpers">
+  private void setChannelOptions(BigtableOptions.Builder builder) throws IOException {
+    setCredentialOptions(builder);
+
+    builder.setRetryOptions(createRetryOptions());
+
+    String channelCountStr = configuration.get(BIGTABLE_DATA_CHANNEL_COUNT_KEY);
+    if (channelCountStr != null) {
+      builder.setDataChannelCount(Integer.parseInt(channelCountStr));
+    }
+
+    // This is primarily used by Dataflow where connections open and close often. This is a
+    // performance optimization that will reduce the cost to open connections.
+    String useCachedDataPoolStr = configuration.get(BIGTABLE_USE_CACHED_DATA_CHANNEL_POOL);
+    if (useCachedDataPoolStr != null) {
+      builder.setUseCachedDataPool(Boolean.parseBoolean(useCachedDataPoolStr));
+    }
+
+    // This information is in addition to bigtable-client-core version, and jdk version.
+    StringBuilder agentBuilder = new StringBuilder();
+    agentBuilder.append("hbase-").append(VersionInfo.getVersion());
+    String customUserAgent = configuration.get(CUSTOM_USER_AGENT_KEY);
+    if (customUserAgent != null) {
+      agentBuilder.append(',').append(customUserAgent);
+    }
+    builder.setUserAgent(agentBuilder.toString());
+  }
+
+  private void setBulkOptions(BigtableOptions.Builder bigtableOptionsBuilder) {
+    BulkOptions.Builder bulkOptionsBuilder = BulkOptions.builder();
```

Comment:
I think you should start with the instance of BulkOptions that already existing in bigtableOptionsBuilder as opposed to create a new instance

Reply:
For this, I had to add getter in `BigtableOptions.Builder`, I have added those changes in this same PR. If you think we should add those as a separate then I can create a PR for that rightaway.